packages/cli/src/executions/parse-range-query.middleware.ts

```diff
@@ -0,0 +1,56 @@
+import * as ResponseHelper from '@/ResponseHelper';
+import type { NextFunction, Response } from 'express';
+import type { ExecutionRequest } from './execution.types';
+import type { JsonObject } from 'n8n-workflow';
+import { ApplicationError, jsonParse } from 'n8n-workflow';
+import {
+	allowedExecutionsQueryFilterFields as ALLOWED_FILTER_FIELDS,
+	schemaGetExecutionsQueryFilter as SCHEMA,
+} from './execution.service';
+import { validate } from 'jsonschema';
+import { BadRequestError } from '@/errors/response-errors/bad-request.error';
+
+const isValid = (arg: JsonObject) => validate(arg, SCHEMA).valid;
+
+/**
+ * Middleware to parse the query string in a request to retrieve a range of execution summaries.
+ */
+export const parseRangeQuery = (
+	req: ExecutionRequest.GetMany,
+	res: Response,
+	next: NextFunction,
+) => {
+	const { limit, firstId, lastId } = req.query;
+
+	try {
+		req.rangeQuery = {
+			kind: 'range',
+			range: { limit: limit ? parseInt(limit, 10) : 20 },
```

Comment:
We might want to put an upper bound to `range` wdyt? I know it's about to fill the gaps but we need to prevent the n8n backend from crashing because a bug in the FE requested too much data.

What about limiting it to 100?